File: frontend/types/analytics.ts
import type { Money } from "@/types/order";

export interface DashboardMetric {
    value: Money | number;
    change_percent: Money;
}

export interface SalesPoint {
    date: string;
    sales: Money;
    orders: number;
}

export interface TopSellingItem {
    menu_item_id: number;
    name: string;
    quantity_sold: number;
    sales: Money;
}

export interface RestaurantDashboardAnalytics {
    sales: DashboardMetric;
    orders: DashboardMetric;
    average_order: DashboardMetric;
    sales_graph: SalesPoint[];
    top_selling_items: TopSellingItem[];
}

File: backend/app/schemas/analytics.py
from decimal import Decimal

from pydantic import BaseModel


class DashboardMetric(BaseModel):
    value: Decimal | int
    change_percent: Decimal


class SalesPoint(BaseModel):
    date: str
    sales: Decimal
    orders: int


class TopSellingItem(BaseModel):
    menu_item_id: int
    name: str
    quantity_sold: int
    sales: Decimal


class RestaurantDashboardAnalytics(BaseModel):
    sales: DashboardMetric
    orders: DashboardMetric
    average_order: DashboardMetric
    sales_graph: list[SalesPoint]
    top_selling_items: list[TopSellingItem]
